### ISSUE TITLE
Accented letters are not accepted in mailer.FROM

### ISSUE THREAD

**Issue by edgar-bonet** [go-gitea/gitea#2188]
- Gitea version (or commit ref): 1.1.2
- Git version: 2.7.4
- Operating system: Ubuntu 16.04.2 LTS
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

I tried to activate gitea's mailer by putting the following in custom/conf/app.ini:

```ini
[mailer]
ENABLED      = true
FROM         = Néel Institut git hosting service <git@***>
USE_SENDMAIL = true
```

When I clicked on the "Send Test Email" button, I got the error message: `Impossible d'envoyer un e-mail de test à 'bonet@grenoble.cnrs.fr': gomail: could not send email 1: gomail: invalid address "=?UTF-8?q?N=C3=A9el_Institut_git_hosting_service_?=": mail: unclosed angle-addr`.

I then replaced "Néel" by "Neel" and everything worked fine, thus I assume this is a problem with gitea (or gomail) mishandling the accented letter "é" (=&nbsp;U+00E9, Latin small letter e with acute).

It should be noted that "`=?UTF-8?q?N=C3=A9el_Institut_git_hosting_service_?=`" is the proper way to encode "`Néel Institut git hosting service `" in an e-mail header, as per [RFC 2047](https://tools.ietf.org/html/rfc2047). Thus, it looks like gomail genuinely attempts to support non-ASCII characters in the e-mail From field.

**Comment by scr34m** [comment]
This is a gomail problem, but there is a workaround, but this problem occurs too when user name contains any non ASCII character.

```go
type mimeEncoder struct {
	mime.WordEncoder
}

bEncoding := mimeEncoder{mime.BEncoding}
name := bEncoding.Encode("utf-8", "József Galamb")

msg := NewMessageFrom([]string{name + " <jon@example.org>"}, "jon@example.org", "subject", "body")
```

If this solution is okay then i could create a PR:
- models/mail.composeIssueCommentMessage for comments notification
- setting.MailService.From any where is used
or simply make a check inside (more general solution) modules/mailer.NewMessageFrom() to field "to" and "from" an parse supplied values with https://golang.org/pkg/net/mail/#ParseAddress

**Comment by zeripath** [comment]
This was fixed by #2559
